test/realm_test.dart

```diff
@@ -1647,3 +1631,13 @@ When newWhen([tz.TZDateTime? time]) {
 extension _IterableEx<T> on Iterable<T> {
   Iterable<T> except(T exclude) => where((o) => o != exclude);
 }
+
+void _writeAndExpectException<T>(Realm realm, String exceptionMessage) {
```

Comment:
I think this should only be used where needed. And I suggest we make it as 
```suggestion
void _writeAndExpectException<T extends Execption>(Realm realm, String exceptionMessage) {
```

Reply:
I used it where it is needed. If you don't like these methods I can copy paste their implementation to all the places where it is needed.

Reply:
Yes, sorry. I got your meaning now. added a suggestion in the comment above.